Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -23,17 +26,48 @@
 #include "MantidKernel/ListValidator.h"
 #include "MantidKernel/Material.h"
 #include "MantidKernel/MersenneTwister.h"
+#include "MantidKernel/PhysicalConstants.h"
+#include "MantidKernel/VectorHelper.h"
 
 using namespace Mantid::API;
 using namespace Mantid::Kernel;
 using Mantid::DataObjects::Workspace2D;
+namespace PhysicalConstants = Mantid::PhysicalConstants;
 
 namespace {
 constexpr int DEFAULT_NPATHS = 1000;
 constexpr int DEFAULT_SEED = 123456789;
 constexpr int DEFAULT_NSCATTERINGS = 2;
 constexpr int DEFAULT_LATITUDINAL_DETS = 5;
 constexpr int DEFAULT_LONGITUDINAL_DETS = 10;
+
+/// Energy (meV) to wavevector (angstroms-1)
+inline double toWaveVector(double energy) { return sqrt(energy / PhysicalConstants::E_mev_toNeutronWavenumberSq); }
```

Comment:
Is there no central place in mantid that stores these unit conversions so we don't have them repeated throughout the code?

Reply:
Good question (especially as I'm normally the cheerleader for the Unit class!)
So the reason I've got these DIY functions at the top of the unit is that this algorithm is loosely based on MonteCarloAbsorption.cpp and that has unit conversion functions at the top (it has toWavelength). But that's just passing the buck...
There is some similar code in Unit.cpp here (see line where a quick conversion to "Momentum" is specified):
```
Energy::Energy() : Unit(), factorTo(DBL_MIN), factorFrom(DBL_MIN) {
  addConversion("Energy_inWavenumber", PhysicalConstants::meVtoWavenumber);
  const double toAngstroms = 1e10;
  const double factor =
      toAngstroms * PhysicalConstants::h / sqrt(2.0 * PhysicalConstants::NeutronMass * PhysicalConstants::meV);
  addConversion("Wavelength", factor, -0.5);
  addConversion("Momentum", 2 * M_PI / factor, 0.5);
}
```
In order to use these quick conversions you have to retrieve the factor and power and then write something like this in your own code `factor * std::pow(x, power)`. So a square root gets implemented as pow(x, 0.5) and I believe this is often slower than the sqrt function:
https://stackoverflow.com/questions/11810686/is-fast-implementation-of-powx-0-5f-faster-than-fast-sqrtx

When I've run this Discus algorithm through the performance profiler the sqrt function has shown up as a pinch point. So I think I'd prefer to leave things as they are rather than make the square roots even slower. But to be honest I've only just researched this. Probably worth a comment next to the DIY functions to justify their use at least.